pkg/clusterversion/clusterversion.go

```diff
@@ -90,3 +99,268 @@ func (builder *Builder) validate() (bool, error) {
 
 	return true, nil
 }
+
+// WithDesiredUpdateImage adds the desired image to the clusterversion struct.
+func (builder *Builder) WithDesiredUpdateImage(desiredUpdateImage string, force bool) *Builder {
+	if valid, err := builder.validate(); !valid {
+		builder.errorMsg = err.Error()
+
+		return builder
+	}
+
+	glog.V(100).Info("Adding the desired image %s to clusterversion %s",
+		desiredUpdateImage, builder.Definition.Name)
+
+	if desiredUpdateImage == "" {
+		glog.V(100).Infof("The desiredUpdateImage is empty")
+
+		builder.errorMsg = "'desiredUpdateImage' cannot be empty"
+
+		return builder
+	}
+
+	builder.Definition.Spec.DesiredUpdate = &v1.Update{Image: desiredUpdateImage, Force: force}
+
+	return builder
+}
+
+// WithDesiredUpdateChannel adds the desired channel to the clusterversion struct.
+func (builder *Builder) WithDesiredUpdateChannel(updateChannel string) *Builder {
+	if valid, err := builder.validate(); !valid {
+		builder.errorMsg = err.Error()
+
+		return builder
+	}
+
+	glog.V(100).Info("Adding the desired updateChannel %s to clusterversion %s",
+		updateChannel, builder.Definition.Name)
+
+	if updateChannel == "" {
+		glog.V(100).Infof("The updateChannel is empty")
+
+		builder.errorMsg = "'updateChannel' cannot be empty"
+
+		return builder
+	}
+
+	builder.Definition.Spec.Channel = updateChannel
+
+	return builder
+}
+
+// Update renovates the existing clusterversion object with the clusterversion definition in builder.
+func (builder *Builder) Update() (*Builder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating clusterversion %s", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("clusterversion object does not exist")
+	}
+
+	builder.Definition.CreationTimestamp = metav1.Time{}
+
+	var err error
+	builder.Object, err = builder.apiClient.ConfigV1Interface.ClusterVersions().Update(
+		context.TODO(), builder.Definition, metav1.UpdateOptions{})
+
+	return builder, err
+}
+
+// Get fetches existing clusterversion from cluster.
+func (builder *Builder) Get() (*v1.ClusterVersion, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting existing clusterversion with name %s from cluster", builder.Definition.Name)
+
+	clusterVersionObj := &v1.ClusterVersion{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name: builder.Definition.Name,
+	}, clusterVersionObj)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to get clusterversion object %s from cluster due to: %w",
+			builder.Definition.Name, err)
+
+		return nil, err
+	}
+
+	return clusterVersionObj, nil
+}
```

Comment:
This method can be removed. Please use `builder.apiClient.ConfigV1Interface.ClusterVersions().Get()` instead

Reply:
removed, thanks